### ISSUE TITLE
Wide Review Comment 2017: Conformance Classes not complete

### ISSUE THREAD

**Issue by tairt** [w3c/webvtt#361]
The following conformance classes are used text but not defined in Section 2.1:

* User agents that don’t support CSS (e.g. Section 7.)
* [...] WebVTT parser [that] requires that the user agent collect WebVTT region settings [..]  

For the later it is implied that a user agent is not always required to collect WebVTT region settings, in which case there will be no region support.

If the first conformance class is added to Section 2.1 this may also require a discussion about the consequences. A user agent that does not support CSS, also does not support text color (see also #279 and #365).

**Comment by dwsinger** [comment]
Yes, User agents that do not support CSS should be added to 2.1. Yes, they should support the base set of colors that we need to decide on.

I think the second sentence is not expressing a conditional requirement, and should be re-phrased. "When the WebVTT parser requires that the user agent collect WebVTT region settings from a string input for a text track, the user agent must run the following algorithm." could be "When the WebVTT parser asks the user agent to collect WebVTT region settings from a string input for a text track, the user agent must run the following algorithm."?

**Comment by tairt** [comment]
> I think the second sentence is not expressing a conditional requirement, and should be re-phrased. 

Yes, possibly this could be made clearer by just replacing the word "require". In the past there was some discussion if regions are mandatory to be supported. So some clarification in this part would help.

**Comment by silviapfeiffer** [comment]
I'm not a fan of "asks the user agent", because that sounds like a network request is made. I'll reformulate slightly differently

**Comment by silviapfeiffer** [comment]
Could you please review.

**Comment by tairt** [comment]
Thanks for the changes. I will review them!

**Comment by tairt** [comment]
Thanks! 

Regarding the new CSS Conformance Class:

In general this looks good to me. There may be one thing to improve:

> All processing requirements in this specification apply, except those in §6 Rendering and in §7 CSS extensions. The user agent must instead implement equivalent visual rendering to what a CSS based rendering produces.

The second sentence is not true for the section 7 because CSS extensions are not supported at all. Section 6 on the contrary is relevant because equivilant rendering needs to be achieved. I think therefore this need to be rephrased. In the text of chapter 6 and 7 this is clear enough but it would be good to already clarify it in the conformance class section.

I did not check yet the support for text color (which is a requirement to have this conformance class), but this is documented in another issue.

Regarding the "WebVTT parser region settings" issue. 

> When the WebVTT parser says to collect WebVTT region settings from a string input for a text track, the user agent must run the following algorithm.

This gets rid of the ambiguity and therefore looks fine to me. You may slightly re-phrase "When the WebVTT parser **algorithm** says...". This would be more in line with how the term "say" is used in the similiar context of the spec.

**Comment by tairt** [comment]
I now reviewed the PR containing the text color solution. It looks good to me but the discussion indicates some missing information how it is linked to this issue. From my side this issue can only be closed once the solution of #365 is agreed.

**Comment by silviapfeiffer** [comment]
I've applied the fixes, but can we please handle the color and the conformance class issues separate from each other? I'd like to land the patch and close this issue.

**Comment by dwsinger** [comment]
#365 is agreed and thus it seems we can close this one.

**Comment by tairt** [comment]
> but can we please handle the color and the conformance class issues separate from each other?

@silviapfeiffer  I understand the intent and in general I agree to keep issues separate. But in this case (at least for us) a solution for color is conditional for having this new conformance class.

The latest commit for PR #389 does not satisfy this requirement. I therefore re-open this issue.

**Comment by tairt** [comment]
Resolved and addressed once PR #389 is merged.
